THREADING.md

```diff
@@ -100,6 +104,13 @@ direct connection to libgcrypt and thus has not convenience functions for
 it (but libgcrypt has macros). Read libgcrypt's
 [threading documentation for more information](http://www.gnupg.org/documentation/manuals/gcrypt/Multi_002dThreading.html)
 
+libssh2 provides a `libssh2_init` function, which is called
+automatically by libssh2 code if needed.  However, `libssh2_init`
+is not thread-safe, and the way in which it is automatically called
+is also not thread-safe, so in a multi-threaded environment you may need
+to call libssh2_init directly to prevent the thread-unsafe automatic
+initialization.
```

Comment:
It's about both, IMO.  Calling `libssh2_init` in a controlled fashion (like from `libgit2_init`) is a lot more thread-safe than letting libssh2 call it automatically when libgit2 uses it for the first time (because the way libssh2 detects whether it's been used for the first time is not thread-safe, and libgit2 could be calling out to it simultaneously from multiple threads - this is exactly what was happening in nodegit/nodegit#907).